### DISCUSSION TITLE
Fork, new project or just create a plugin for DNS Management w/Wildcard SSL's support provided.

### DISCUSSION THREAD

**Discussion by maietta** [caprover/caprover#1403] [Ideas]
I am looking into building out a solution to automate DNS management for exposed Caprover apps. I am weighing out the different approaches I can take to provide a nice solution.

My solution would entail the following features:
- Automated DNS management of hostname exposed Apps.
- Wildcard SSL's by default on naked domains. (i.e., premoweb.com)
- For multi-node swarm environments, allow for multiple discoverable ingress IP's to be assigned per A or AAAA records.
- Assign one or more apps as designated for "domain parking" purposes.
- DNS Zone cleanup dialog offering warning of non A & AAAA records as well as an option to park domain with a selector.
- Support for many 3rd party DNS providers including Vultr, Epik, StackPath, Route53 and more.

As a one-click-app, this functionality could not yet exist in it's entirety. Actions taken on apps in the API can't be seen by the one-click-app and so it would seem to me that designing a one-click-app is a fundamentally flawed approach.

Maintaining a separately maintained fork of Caprover seems to be completely overkill and I certainly do not want to take away from the hard work of Caprover's authors and other contributors. It would be amazing to see continued contributions to Caprover and it's one-click-apps ecosystem.

It almost seems like I need to start a new PaaS inspired by Caprover because of the following:

- I would like to also swap out Nginx as the reverse proxy in favor of Traefik
- I would like to see multi-user environment with ACL's for a variety of actions, which would be a moon-shot feature set, having the potential to open this solution up as a possible self-serve hosting platform.

The thought of all this almost paralyzes because of the amount of work that would be required to bring this to fruition.

I would absolutely love to hear other people's thoughts on this, especially from the main contributors of Caprover.

**Comment by githubsaturn** [comment]
> It almost seems like I need to start a new PaaS inspired by Caprover because of the following:

On top of what you mentioned, you can also swap Swarm for K8. Overall, yes, based on your requirement, I think kicking off a new project inspired by CapRover is probably the best best. 

If you are only interested in DNS management, you can write a completely standalone app that performs all this and just communicate witih CapRover's api to fetch the apps and update them if needed.

**Comment by maietta** [comment]
Thanks for the feedback, @githubsaturn.

I'm very much leaning toward creating a whole new project after some careful thought and discussion with others, including your feedback.

I am interested in much more than DNS management. Cloud firewall management, no-downtime between deploys, automated backups and configuration export/imports are some of the things I want to include.

Instead of working on cars this spring and summer, I think I'll be doing this.
